### PR TITLE
Create Term Entry for Python String function `.endswith()`

### PR DESCRIPTION
**Pull request by Pandz18**
### Description

Created an entry for the in-built python function, `.endswith()`.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by Pandz18** [comment]
Also a question, I had linked my GitHub profile to my codecademy account but for the reset submission, it still shows as anonymous contributor.

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (KTom101) [inline content/python/concepts/built-in-functions/terms/endswith/endswith.md:62]: Changing to reflect added index values.
```suggestion
# Output: True
```

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (KTom101) [inline content/python/concepts/built-in-functions/terms/endswith/endswith.md:26]:
```suggestion

Checking for the value `g` at the end of the string, `example`:

```

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (KTom101) [inline content/python/concepts/built-in-functions/terms/endswith/endswith.md:36]:
```suggestion

Checking for the value `s`:

```

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (KTom101) [inline content/python/concepts/built-in-functions/terms/endswith/endswith.md:56]: Let's change this example to have a starting and ending index.
```suggestion

The following checks for `st` and has a starting index of `5` and an ending index of `12`:

```

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (KTom101) [inline content/python/concepts/built-in-functions/terms/endswith/endswith.md:46]:
```suggestion

Checking for the value `ring`:
```

**Review by KTom101 — COMMENTED** [review]
Hello @Pandz18, I've completed my review for this entry.  Please take a look at these changes and let us know when you're done making revisions so we can move this on to the 2nd review. Thanks!

File: content/python/concepts/built-in-functions/terms/endswith/endswith.md
Comment (Pandz18) [reply content/python/concepts/built-in-functions/terms/endswith/endswith.md:55]: @KTom101 Hey, just updated the file, hope it's fine now!

**Review by KTom101 — APPROVED** [review]
@Pandz18 Great! This is ready for a second review! 
@Dusch4593 @yangc95

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:2]:
```suggestion
Title: '.endswith()'
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:17]:
```suggestion
The `.endswith()` method returns `True` if a particular value is attached to the end of a given string. Otherwise, it returns `False`.
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:20]:
```suggestion

With a given `string`, the `.endswith()` method can be used in the following way:

```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:29]:
```suggestion
The `value` is required argument that is checked for whether it exists at the beginning of the `string`. 

Conversely, the `value` can be tested whether it is the end of a particular substring of `string` by using the optional `start` and `end` index arguments.
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:81]:
```suggestion
## Codebyte example
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:83]:
```suggestion
```codebyte/py
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:88]:
```suggestion
example_A = "This is a string"
check_A = example_A.endswith("ring")
print(check_A)

// .endswith() is case-sensitive
example_B = "This is a string"
check_B = example_B.endswith("G")
print(check_B)
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:65]: I think we should go with just three non-codebyte example. Could we delete lines #55-65 here?

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:67]:
```suggestion
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:57]:
```suggestion
The next example returns `True` after checking for the value `ring`:
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:33]:
```suggestion
## Examples

The following ckecks for the value `g` at the end of the `example` string:
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:45]:
```suggestion
This example returns `False` after checking for the value `s`:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Hey @Pandz18! Thank you for this latest entry! In c71a57aa81803f4ff82e622c4e5a0915b807797d, I moved the term folder + file from the `python/concepts/built-in-functions/terms` directory to `python/concepts/strings/terms` directory because `.endswith()` is technically a String method.

But other than that, this entry is off to a solid start! 💪 I went ahead and left comments/suggestions for you to look over. After you're finish pushing the next batch of changes, tag me so we can proceed onward. 😃

**Comment by Pandz18** [comment]
Hey, @Dusch4593, sorry for the delay, so here's the update for the readme:
- Deleted the extra example and removed all the headings example under a common one , called examples.
- made changes to the wording.

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:59]:
```suggestion
## Codebyte Example
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:61]:
```suggestion
```codebyte/python
```

Comment (Dusch4593) [reply content/python/concepts/strings/terms/endswith/endswith.md:61]: For regular snippets, let's use \```py and for Codebytes let's use \```codebyte/python

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:3]:
```suggestion
Description: 'Checks whether or not a string ends with a given value.'
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:17]:
```suggestion
The `.endswith()` method checks a value against a given string and returns `True` if the value exists at the end of that string. Otherwise, it returns `False`.
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:67]:
```suggestion
```

Let's avoid using comments in Codebytes whenever possible (this one was already mentioned earlier in the entry).

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:22]: Was there a code snippet that was supposed to go here (using \```pseudo)? Something like this?

```pseudo
string.endswith(value, start, end)
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:25]:
```suggestion
Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string`. This is done within the `start` and `end` index arguments.
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:30]: I think it would. be better to consolidate these snippets into two: 

* One code snippet for the Python. \```py
* One output snippet for the printed boolean values. \```shell
* After the snippets should be 2-3 sentences explaining how each `check_` gets its value.

```py
example_str = "This is a string"

check_A = example_str.endswith("g")
check_B = example_str.endswith("s")
check_C = example_str.endswith("st", 5, 12)

print("A: ", check_A)
print("B: ", check_B)
print("C: ", check_C)
```

And the output would be:

```shell
A: True
B: False
C: True
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:27]:
```suggestion
## Example
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:21]:
```suggestion
Given a `string`, the `.endswith()` method can be used in the following way:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Pandz18 Hey there! Thanks for the changes 😄 I left some comments/suggestions for you to look through. This is shaping up to be a pretty solid entry! 🚀 🙏

**Review by yangc95 — COMMENTED** [review]
@Pandz18 I have also gone ahead and left some minor comments/suggestions.

Please have a look on your available time so we can get this entry up 👌🏻

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (yangc95) [inline content/python/concepts/strings/terms/endswith/endswith.md:29]: I think these read better as bullet points but I'm wondering about the general opinion on this @Dusch4593 @KTom101 @Pandz18 

If we decide to keep this in paragraph form, can we edit some of the wording?
```suggestion
The required `value` argument is checked if it exists at the end of the `string`.  It is also case-sensitive.

Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string` using the `start` and `end` index arguments.
```

Comment (KTom101) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: @yangc95 I think it can be condensed a little and changed into bullet points. Something like below maybe? 🤔

```suggestion
-`value` is a required, case-sensitive argument that is checked for at the end of the `string`.

- `value` can also be tested with the optional `start` and `end` index arguments to test whether it is at the end of a particular substring within the `string`.
```

But I think it could also be kept as a paragraph, reworded, and remove line 25 which makes it sound like a list is going to be followed after.

Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: @yangc95 I would also think we can change it to a list. I can try to re-word it and will update it as soon as I can!

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:43]:
```suggestion
```
```

Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:43]: Let's end the \```py code snippet here, add a new line, and then begin a new \```shell output snippet with the output lines (except for "# Output: ")

```
A: True
B: False
C: True
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [reply content/python/concepts/strings/terms/endswith/endswith.md:32]: Just looked at it and it's looking better! 👍 I left one more suggestion for the snippet below.

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/endswith/endswith.md:49]:
```suggestion
From the above example, for `check_A` the output is True as the `.endswith()` function checks whether the the string `example_str` ends with the character "g".
```

When referencing a piece of code inline, let's surround with just one backtick on each side. Three backticks should be used only for code snippets that span multiple lines.

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (yangc95) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: Cool, thanks for the input!

I also think line 25 can be improved upon by mentioning something about the parameters being passed in rather than just "using" the method in certain ways. Alternatively (which I agree with) we can also remove that line as @KTom101 suggested.

If line 25 is removed, I think the rest can stay in paragraph format 👌🏻

**Comment by Pandz18** [comment]
> @Pandz18 Hey there! Thanks for the changes 😄 I left some comments/suggestions for you to look through. This is shaping up to be a pretty solid entry! 🚀 🙏

Thank you @Dusch4593 ! Will definitely go through them!

**Comment by Pandz18** [comment]
> @Pandz18 I have also gone ahead and left some minor comments/suggestions.
> 
> Please have a look on your available time so we can get this entry up 👌🏻

Hi @yangc95 , will go through them!

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: > Cool, thanks for the input!
> 
> I also think line 25 can be improved upon by mentioning something about the parameters being passed in rather than just "using" the method in certain ways. Alternatively (which I agree with) we can also remove that line as @KTom101 suggested.
> 
> If line 25 is removed, I think the rest can stay in paragraph format 👌🏻

If it's fine with you, I can try to re-word it as a list and if you feel it would be better, leave it as is.

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md]: > When referencing a piece of code inline, let's surround with just one backtick on each side. Three backticks should be used only for code snippets that span multiple lines.

Understood, thank you!

**Comment by Pandz18** [comment]
@yangc95 @Dusch4593 @KTom101   I've updated the file,  hope it's better now !

**Review by Dusch4593 — APPROVED** [review]
@Pandz18 Looks solid! 👍 

I think this is ready to merge, but I made a small edit in 9916117c422894f3081514b2baa2abaedc19ebee where I added a transition sentence in between the code snippets. Let's always make sure there is text between the snippets. 😄

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (yangc95) [inline content/python/concepts/strings/terms/endswith/endswith.md:53]:
```suggestion
From the above example, for `check_A` the output is `True` as the `.endswith()` function checks whether the `example_str` string ends with the character "g".
```

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (yangc95) [inline content/python/concepts/strings/terms/endswith/endswith.md:29]: Still thinking that we don't need those back ticks for string because we're referring to it as some value's data type rather than a required code component. Wondering about everyone's opinion on this @Dusch4593, @KTom101

**Review by yangc95 — COMMENTED** [review]
Caught some things! Also added something for discussion

File: content/python/concepts/strings/terms/endswith/endswith.md
Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: > Still thinking that we don't need those back ticks for string because we're referring to it as some value's data type rather than a required code component. Wondering about everyone's opinion on this @Dusch4593, @KTom101

I agree that we do not need them, my bad for that will update the file.

Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md:29]: @yangc95 is it fine now?

Comment (yangc95) [reply content/python/concepts/strings/terms/endswith/endswith.md]: Perfecto 👌🏻 

Also, it's all good 😊 For context, we made a similar change in another entry. Just making sure it is consistent.

Comment (yangc95) [reply content/python/concepts/strings/terms/endswith/endswith.md]: Oh shoot, but can we place those back on the value, start, and end keywords?

Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md]: > Oh shoot, but can we place those back on the value, start, and end keywords? @Pandz18

Yeah sure, sorry bout that.

Comment (Pandz18) [reply content/python/concepts/strings/terms/endswith/endswith.md]: > Perfecto 👌🏻
> 
> Also, it's all good 😊 For context, we made a similar change in another entry. Just making sure it is consistent.

I understand 🙂

**Comment by Dusch4593** [comment]
How is everyone feeling about this PR? Anything left to check before we merge?

cc: @Pandz18 @yangc95 @KTom101

**Review by yangc95 — APPROVED** [review]
Looks ready for merge!

**Comment by Dusch4593** [comment]
Congrats @Pandz18 for contributing another solid entry to Docs! 🚀 🎉 🎊 When the entry is live, it can be found here: https://www.codecademy.com/resources/docs/python/strings/endswith

**Comment by Pandz18** [comment]
> Congrats @Pandz18 for contributing another solid entry to Docs! 🚀 🎉 🎊 When the entry is live, it can be found here: https://www.codecademy.com/resources/docs/python/strings/endswith

Thank you so much @Dusch4593 @yangc95 @KTom101 !! 🚀 🎉

**Comment by Dusch4593** [comment]
@Pandz18 While I will always say that experience is the absolute best teacher, there are definitely some resources out there that I've found particularly helpful when learning to write good technical content/documentation:

- [Write the Docs](https://www.writethedocs.org/) - for learning about possible conferences/meetups and connect with others online through Twitter and Slack!
- [Google's courses on Tech Writing](https://developers.google.com/tech-writing/overview) are also pretty good, in my opinion. They really helped me with understanding how to better think like a tech writer and what to look for when reviewing/writing content.

**Comment by yangc95** [comment]
> If y'all wouldn't mind be asking, I want to become better at technical writing, what would y'all suggest that I do? Should I read/watch articles/courses/videos for guidance or learn from my contributions on here? 🙂

@Pandz18 That's awesome to hear!

+1 on @Dusch4593's recommendation. It's important to stay connected with the community. Also, I might check out those courses on technical writing from Google as well 👀 

💡 I'll first recommend Codecademy courses first because all the courses break down a topic into digestible chunks. If you're learning a particular language or tool with some goal in mind, try following the thought process of how each part of the lesson breaks down a certain point, and try to not leave any rocks unturned if you're unsure about the way something is explained. Try explaining something in your own words and write those down as notes if writing things down helps aid in your learning!

I really love tutorials as well, and when they're in documentation format like this one from Apple Developer (Intermediate/advanced) on [SwiftUI Essentials](https://developer.apple.com/tutorials/swiftui#swiftui-essentials), it's fun going through it ☺️

**Comment by KTom101** [comment]
@Pandz18 To add to what @yangc95 and @Dusch4593 have said, it's a good idea to have someone with no technical experience read your content. Pick out concepts that a beginner could understand and write about them in your own words. Share it with someone who has little to no knowledge in the subject and see if they can follow along. 

Also, from my experience, I found tutoring others helps with technical writing too (though it may seem unrelated). It really allows you to see what you can work on when explaining things. You can take note of people's reactions and figure out what works and what doesn't. You can even write out detailed lesson plans that you could use to tutor with. In the past, I've used lesson plans to explain concepts to myself first before explaining them to another person. 😄

**Comment by AaronJohnson02** [comment]
Hey, i apologise for the late reply, thank y'all so much for the advice. I will definitely try out the codecademy and Google courses whenever possible and also will definitely try to teach my peers to improve the skill.

**Pull request merged**
